### PR TITLE
BUG: fix to_timestamp out_of_bounds

### LINKED ISSUE #19643: Period.asfreq issue caused(?) by C division rules
This behavior is present both before and after #19608/#19540.

```
per = pd.Period('0001-01-01', freq='B')
>>> per + 1
Period('0001-01-02', 'B')
>>> per - 1
Period('0001-01-05', 'B')
>>> per - 4
Period('0001-01-02', 'B')
>>> per - 5
Period('0000-12-25', 'B')
```

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #19643
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/scalar/period/test_asfreq.py [hunk 1]
```diff
@@ -30,9 +30,6 @@ def test_asfreq_near_zero_weekly(self):
         assert week1.asfreq("D", "E") >= per1
         assert week2.asfreq("D", "S") <= per2
 
-    @pytest.mark.xfail(
-        reason="GH#19643 period_helper asfreq functions fail to check for overflows"
-    )
     def test_to_timestamp_out_of_bounds(self):
         # GH#19643, currently gives Timestamp('1754-08-30 22:43:41.128654848')
```
Comment (jorisvandenbossche) [inline pandas/tests/scalar/period/test_asfreq.py]: you can remove this comment as well I think?

Comment (jbrockmendel) [reply pandas/tests/scalar/period/test_asfreq.py]: updated

**Review by jreback — CHANGES_REQUESTED** [review]
i guess this is a bug fix, can you add a note, ok for 0.25.1 i think.

**Comment by jbrockmendel** [comment]
added whatsnew, found this also fixes the analogous bug in PeriodIndex, need to add test for that

**Comment by jbrockmendel** [comment]
updated with additional test

**Comment by jbrockmendel** [comment]
@jorisvandenbossche gentle ping

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**